src/rgw/rgw_sal_sfs.h

```diff
@@ -422,6 +424,21 @@ class SFStore : public Store {
   std::string get_cls_name() const { return "sfstore"; }
 };
 
+struct ScopedStoreBucketRefresher {
+
+  ScopedStoreBucketRefresher(SFSBucket &bucket)
+  : store(bucket.get_store()),
+    meta_buckets(sfs::get_meta_buckets(store.db_conn)),
+    db_bucket(bucket.get_info(), bucket.get_attrs()) {}
+
+  ~ScopedStoreBucketRefresher(){
+    store._refresh_buckets_safe();
+  }
+
+  SFStore &store;
+  sfs::MetaBucketsRef meta_buckets;
+  sfs::sqlite::DBOPBucketInfo db_bucket;
+};
```

Comment:
1) 

> If the idea is to create the db bucket, store it and refresh... could just a private method in SFSBucket do the work?

This technique/pattern is very old in C++ and it is universally known as [RAII](https://en.cppreference.com/w/cpp/language/raii).
Basically, why it is employed? because you want that when a certain object leaves its scope, something happens.
It could be release of an allocated memory, release of an acquired lock or everything you want to do at the end of a certain scope.
Why in C++ this is especially convenient? Because, no matter what is causing a scope to be leaved (return, throw): the compiler ensures the destructor code the be called (for stack allocated objects).
`ScopedStoreBucketRefresher` is convenient when you want  `_refresh_buckets_safe` to be called at the end, no matter what.
If someone at a certain point introduces a `return` in the middle of a method he has not to care to add the call to `_refresh_buckets_safe` for that path, that call is guaranteed by the compiler. 

2) 

> I mean, it forces to make the SFSBucket store available via newly added get_store() (no big problem).

This is intended, the user should not be able to choose what store to pass to the constructor.
This is a behavior constrained for this type, the store is always taken from the SFSBucket argument.

Btw, nothing is written in the stone here, if you think there are motivations to avoid the usage for this pattern we can discuss by voice. 
Also @jecluis raised some doubts regarding forcing the code to take the store from the SFSBucket argument; so maybe I'm not seeing all the implications.

Reply:
Sorry @0xavi0 I know you know, but better be clear to avoid incorrect interpretations :)
I think there is something unclear here: after an operation that is persisting something on the sqlite table, eg: `store_bucket` is it mandatory to call `_refresh_buckets_safe` ? 
I'm assuming `_refresh_buckets_safe` must always called to keep aligned memory structures with the content of the database (but I could have not properly understood when it needs to be called).
If this assumption is true, I think `ScopedStoreBucketRefresher` has a dignity to exist and I can be useful to avoid a lot of bug related to `_refresh_buckets_safe` forgetfulness.

Reply:
@jecluis a penny for your thoughts :)

Reply:
Well, given the usage of this class, I don't see a particular clear benefit in it existing. I think it's adding unnecessary complexity to issue a bucket refresh.

Reply:
Sorry, I wrote my last comment replying to @0xavi0 with the page not updated.
Ok, I'm in minority here ;) so tomorrow I will get rid of  this class.
Thanks both for your comments.

Reply:
>  Current code btw is not checking sqlite operations, if something at that level happens I think the process simply crashes. We should introduce `try..catch` blocks and check for exceptions. In case of exception we should return and not call `_refresh_buckets_safe`.

yeah, there's already a bug opened for that. 
https://github.com/aquarist-labs/s3gw/issues/4
I would prefer to deal with the exceptions in the `sqlite` interface to avoid tons of `try... catch` in the code using it. But I need to think on how to do it...